File: backend/strategies.py
from fastapi import APIRouter
from pydantic import BaseModel
import json
import os

strategies_router = APIRouter()
STRATEGY_PATH = os.path.join("config", "strategies.json")

class StrategyItem(BaseModel):
    name: str
    condition: str
    winRate: float
    enabled: bool
    priority: int

@strategies_router.get("/strategies")
async def get_strategies():
    if not os.path.exists(STRATEGY_PATH):
        return [
            {
                "name": "추세선 돌파",
                "condition": "3번 터치 후 볼밴 확장",
                "winRate": 70.2,
                "enabled": True,
                "priority": 1
            },
            {
                "name": "RSI 다이버전스",
                "condition": "쌍바닥 + 캔들 반전",
                "winRate": 65.8,
                "enabled": False,
                "priority": 2
            }
        ]
    with open(STRATEGY_PATH, "r", encoding="utf-8") as f:
        return json.load(f)

@strategies_router.post("/strategies/update")
async def update_strategy(item: StrategyItem):
    os.makedirs("config", exist_ok=True)
    strategies = []
    if os.path.exists(STRATEGY_PATH):
        with open(STRATEGY_PATH, "r", encoding="utf-8") as f:
            strategies = json.load(f)

    for i, s in enumerate(strategies):
        if s["name"] == item.name:
            strategies[i] = item.dict()
            break
    else:
        strategies.append(item.dict())

    with open(STRATEGY_PATH, "w", encoding="utf-8") as f:
        json.dump(strategies, f, indent=2, ensure_ascii=False)

    return {"success": True}

File: frontend/app/lib/data/strategies.ts
export type Strategy = {
  id: string
  name: string
  winRate: number
  successCount: number
  priority: number
  description: string
  status: '사용 중' | '비활성화'
  condition: string
  lastUpdated: string
}

export const strategies: Strategy[] = [
  {
    id: "s1",
    name: "추세선 돌파",
    winRate: 70.2,
    successCount: 31,
    priority: 1,
    description: "세 번 터치 후 볼린저밴드 확장에 따라 롱 진입.",
    status: "사용 중",
    condition: "3번 터치 후 볼밴 확장",
    lastUpdated: "2025-08-03"
  },
  {
    id: "s2",
    name: "RSI 다이버전스 쌍바닥",
    winRate: 65.8,
    successCount: 24,
    priority: 2,
    description: "RSI 하단 다이버전스와 쌍바닥 구조 + 캔들 반전 발생 시 진입.",
    status: "비활성화",
    condition: "RSI 다이버전스 쌍바닥 + 캔들 반전",
    lastUpdated: "2025-07-29"
  }
]
